docs/design/reference-values.md

```diff
@@ -0,0 +1,95 @@
+# Reference values data flow
+
+## Overview
+
+Attestation in Confidential Clusters is based on PCR values reported by a TPM.
+These values can be predicted when the exact OS the node is expected to be used is known, by means of the [compute-pcrs](https://github.com/confidential-clusters/compute-pcrs) library.
+In the design of Confidential Clusters, the OS is represented by a bootable container image with a UKI.
+
+This document describes how a bootable image tag becomes approved and revoked, and how the set of approved image tags is turned into reference values to be used by Trustee's [reference value provider service](https://github.com/confidential-containers/trustee/tree/main/rvps).
+
+## Dual source for approved images: MachineConfigs & kubectl interaction
+
+In OpenShift, updates to nodes are defined using Kubernetes resources.
+The `MachineConfig` CR can be applied to e.g. define the bootable image reference which nodes classified by some selector should use.
+Target machine configs are then referenced to in `MachineConfigPools`.
+Because such explicit updates are assumed to be intended by the cluster administrator, the Confidential Clusters operator can watch the MachineConfigPools and set the images that they reference as approved.
+
+However, kubectl interaction is also supported, both for avoiding reliance on OpenShift and for manual intervention.
+
+## Split data store: CRD for approved images, ConfigMap for PCR parts
+
+For better interaction with kubectl, approved images are specified as a very simple custom resource:
+
+```yaml
+apiVersion: confidential-containers.io/v1alpha1
+kind: ApprovedImage
+metadata:
+  name: my-scos
+  namespace: confidential-clusters
+spec:
+  reference: quay.io/my-registry/scos-kernel-layer
```

Comment:
I'm wondering if we should specifically reference to a an image and its sha here. If we use tags, it can be different images pushed under the same tag. Can this be a problem?

Reply:
yea this is an issue. Previously, in a comment merged in #14, I postulated
> Non-goal: Support tags whose referenced versions change (e.g. `latest`). This would introduce hard-to-define behavior for disallowing older versions that were introduced as a tag that is still allowed.

which was an easy way of supporting tags for convenience and saying "re-using tags is unsupported, deal with it". Indeed, it would be safer and still relatively user-friendly (especially when using MachineConfig) to support only the SHA. LMK what you think.

Reply:
Yes, I think for now we should support only the shas